### DISCUSSION TITLE
Can Aggressive Caching be enabled per collection

### DISCUSSION THREAD

**Discussion by jon-m-bottoms** [ravendb/ravendb#16590] [Q&A] [answered]
I have a message handler that processes 100s of events a minute and effectively does two things:
* loads config information (static, rarely changes) to decide how to process the message
* loads record for processing 

I would like to aggressively cache the config data, but there is no point trying to cache other data because it's always expected to be changed.

I presume I can manually implement this functionality with the Changes API, just want to make sure I wasn't missing any options in the caching config before I started.

Thanks

**Comment by ayende** [comment, marked as answer]
You can enable aggressive caching only for the portions where you want it to, instead of globally.
Aggressive caching works at the request level, so not actually aware of what you are working on.

**Reply by jon-m-bottoms** [reply]
ok, that's fine, but I can't seem to get aggressive caching to work globally.

If I enable it per session it works as expected, in that the Traffic Watch only shows the first request. 
`
using(_session.Advanced.DocumentStore.AggressivelyCache())
`
However, if I attempt to enable it at the global level
` documentStore.Initialize();
 documentStore.AggressivelyCache();`


my session (dependency injected or manually created from the documenstore), never retrieves the previous requests from cache
` using (IAsyncDocumentSession _session = _store.OpenAsyncSession())`

In fact, the RequestExecutor.AggressiveCaching value is null in this case.

From the documentation my understanding was that if aggressive caching was enabled at the documentstore level, I wouldn't need to specify it's use for each session.

Please let me know if I'm missing something or not understanding the feature correctly,

**Reply by ayende** [reply]
You cannot enable that globally, that is meant to be _scoped_ for the local async / thread operation.

**Reply by ReneB** [reply]
@ayende the [documentation](https://ravendb.net/docs/article-page/5.4/csharp/client-api/how-to/setup-aggressive-caching) suggests differently - is the documentation incorrect?

**Reply by ayende** [reply]
The reference for "globally" there is meant to refer to cross sessions, The way this works, it sets a flag, basically, in the current thread /async context. So any session created within that scope is covered.

It is not applied globally to all sessions (that would make threading / concurrency issues) and no allow you to create different scopes

**Comment by jon-m-bottoms** [comment]
Understood. thanks
